serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/integration-tests-with-postgresql.adoc

```diff
@@ -1 +1,212 @@
+details see link:{quarkus_testing_guide_integration_test_url}[@QuarkusIntegrationTest] guide.
+* *Quarkus Dev Services* facilitates writing integration tests that need launching services to support the application. More explained in link:{quarkus_testing_guide_launching_containers_url}[Testing Quarkus application - launching containers] guide.
+
+*@QuarkusIntegrationTest* works out of the box with the additional containers launched via link:{quarkus_dev_services_url}[*Quarkus Dev Services*].
+
+[[proc-test-sw-application-postgresql]]
+== Testing a Serverless Workflow application with PostgreSQL persistence
+
+.Prerequisites
+include::../../pages/_common-content/getting-started-requirement.adoc[]
+* Enable the application persistence using PostgresSQL. More details in {persistence_with_postgresql_guide}[Running Serverless Workflow service using PostgresSQL] guide.
+* Docker is installed.
+* link:{postgresql_url}[PostgreSQL] is installed. For information about PostgreSQL installation and configuration, see link:{postgresql_doc_url}[PostgreSQL documentation].
+
+[NOTE]
+====
+Although PostgreSQL installation is mentioned as a requirement, this guide relies on running it as a Docker service instead.
+====
+
+.Procedure
+. Add the required test dependencies to the `pom.xml` file of your Kogito Serverless Workflow application:
++
+--
+
+.Dependencies required for HTTP-based testing in JVM mode.
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-junit5</artifactId>
+    <scope>test</scope>
+</dependency>
+<dependency>
+    <groupId>io.rest-assured</groupId>
+    <artifactId>rest-assured</artifactId>
+    <scope>test</scope>
+</dependency>
+----
+
+.Awaitility
+[source,xml]
+----
+<dependency>
+  <groupId>org.awaitility</groupId>
+  <artifactId>awaitility</artifactId>
+  <scope>test</scope>
+</dependency>
+----
+
+`Awaitility` dependency allows the test to express time expectations of an asynchronous system.
+For more information, see link:{awaitility_url}[Awaitility] website.
+
+--
+
+. Check PostgreSQL resources dependencies
++
+--
+The link:{quarkus_database_dev_services_url}[*Quarkus Dev Services for Databases*] will be enabled when a reactive or JDBC datasource extension is present in the application.
+Ensure the Application's `pom.xml` has the dependency to the needed PostgreSQL resources:
+
+.JDBC persistence add-on
+[source,xml]
+----
+<dependency>
+  <groupId>org.kie.kogito</groupId>
+  <artifactId>kogito-addons-quarkus-persistence-jdbc</artifactId>
+</dependency>
+----
+
+.Quarkus JDBC PostgreSQL
+[source,xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-jdbc-postgresql</artifactId>
+</dependency>
+----
+
+.Quarkus Agroal Data Source
+[source,xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-agroal</artifactId>
+</dependency>
+----
+--
+
+. Enable Quarkus Database Dev services
```

Comment:
@nmirasch I think here we should instead link to Quarkus. I think this can get outdated easily as Quarkus changes things, so we just need to tell users to make sure they are aware of the conditions, wdyt?

Reply:
makes sense!